core/src/main/java/org/pentaho/platform/engine/core/system/objfac/AbstractSpringPentahoObjectFactory.java

```diff
@@ -203,12 +203,11 @@ protected Object retrieveViaSpring( String beanId ) throws ObjectFactoryExceptio
 
       previousSpringSession = SpringScopeSessionHolder.SESSION.get();
 
-      if ( session instanceof StandaloneSession ) {
-        // first ask Spring for the object, if it is session scoped it will fail
-        // since Spring doesn't know about StandaloneSessions
+      // Save the session off to support Session and Request scope.
+      // If it is session scoped it would fail without this, since Spring doesn't know about StandaloneSessions.
+      SpringScopeSessionHolder.SESSION.set( session );
```

Comment:
Setting the `SpringScopeSessionHolder.SESSION.set( session )` to the current session is actually required by all kinds of sessions, standalone or not.